src/router/imageProcessRouter.ts

```diff
@@ -44,13 +50,25 @@ router.post(
           json
         );
 
-        imageResult = await uploadImageToBucket(
-          response.data.images[0],
-          prompt,
-          imageId
+        const isSafe = await isContentSafeForDisplay(
+          model, // Model to prevent multiple initialization
+          response.data.images[0]
         );
 
-        imageId = imageResult._id;
+        if (!isSafe && imageId) {
+          await removePromptModel(imageId)
+          return res.status(200).json({ Message: 'Content is not safe for display' })
+        } else if (!isSafe) {
+          return res.status(200).json({ Message: 'Content is not safe for display' })
```

Comment:
```suggestion
        if (!imageId) {
          await removePromptModel(imageId)
        } 
        
        if (!isSafe) {
          return res.status(200).json({ Message: 'Content is not safe for display' })
```

Something like this is easier to read?